### PR TITLE
(DOCSP-23555): Update Data Lake -> Data Federation

### PR DESCRIPTION
**Pull request by nlarew**
### Jira

- (DOCSP-23555): Update Data Lake -> Data Federation

### Staged Changes (Requires MongoDB Corp SSO)

- [Staged](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/atlas-data-federation/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-app-services/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/functions/context.txt [hunk 1]
```diff
@@ -154,7 +154,9 @@ values through the ``context.environment`` interface.
 Connect to a MongoDB Data Source or Third-Party Service (``context.services``)
 ------------------------------------------------------------------------------
 
-You can access a client for a linked MongoDB Atlas cluster or Data Lake through the ``context.services`` interface. You can also access third-party services, although this feature is deprecated. 
+You can access a client for a linked MongoDB Atlas cluster or
+{+adf-datasource+} through the ``context.services`` interface. You can
+also access third-party services, although this feature is deprecated.
```
Comment (MongoCaleb) [inline source/functions/context.txt:159]: I wonder if, by this point, we should just leave off this last sentence? How long has it been deprecated now?

**Review by MongoCaleb — APPROVED** [review]
LGTM % 2 comments

File: source/functions/context.txt (hunk 1 above)
Comment (nlarew) [reply source/functions/context.txt:159]: I think that we should keep it until we fully deprecate services given that this is a reference page. Right now they're soft-deprecated, where existing services still work.

File: source/manage-apps/configure/config/data_sources.txt [hunk 2]
```diff
@@ -80,9 +80,9 @@ MongoDB Clusters
      - If ``true``, clients may :ref:`connect to the app over the MongoDB Wire
        Protocol <wire-protocol-connect>`.
 
-.. _config-datalake:
+.. _config-adf
 
-MongoDB Data Lakes
+{+adf-instance+}s
```
[the remark this answers is not in the document]
Comment (nlarew) [reply source/manage-apps/configure/config/data_sources.txt:85]: I think it's actually the ref link above the heading, but added some tildes here too

**Pull request merged**